azure_arc_k8s_jumpstart/aro/aro_connect.sh

```diff
@@ -213,13 +213,27 @@ echo "adminPassword=$adminPassword"
 echo "apiServer=$apiServer"
 echo "done"
 
-sleep 10s
+sleep 30s
 
 # Log into the OC command
 echo "==============================================================================================================================================================="
 oc login $apiServer -u $adminUser -p $adminPassword
 echo "==============================================================================================================================================================="
 
+echo "Upload kubeconfig to blob."
+storageName="arcstoragearo$RAND$RAND"
+if [ "$($az storage account check-name --name $storageName --query nameAvailable -o tsv 2>/dev/null)" = "true" ]; then
+    $az storage account create --name $storageName --resource-group $RESOURCEGROUP --location $LOCATION --sku Standard_ZRS -o none 2>/dev/null
+    $az storage container create --account-name $storageName --name arccontainer --public-access blob -o none 2>/dev/null
+    $az storage blob upload --account-name $storageName --container-name arccontainer --name config --auth-mode key --file /root/.kube/config -o table
+    $az group update -n $RESOURCEGROUP --tag currentStatus=kubectlUploaded fileurl="https://${storageName}.blob.core.windows.net/arccontainer/config" aroAdminUser=$adminUser aroAdminPassword=$adminPassword aroUrl=$apiURL 2>/dev/null
+    echo "File Uploaded."
+else
+    $az group update -n $RESOURCEGROUP --tag currentStatus=kubectlFailed 2>/dev/null
+    echo "File Upload Failed"
+fi
+echo "done."
+
 if [ ! "$($az connectedk8s show -g "$RESOURCEGROUP" -n "$ARC" --query provisioningState -o tsv 2>/dev/null)" = "Succeeded" ]; then
```

Comment:
I don't see any implementation for the Custom Location ObjectID. See example [here](https://github.com/microsoft/azure_arc/blob/main/azure_arc_k8s_jumpstart/cluster_api/capi_azure/arc_capi_azure.sh#L166). 

This is required for supporting future Azure Arc services that will be deployed on the cluster using SPN-based automation.

Reply:
FYI we probably should not use custom-locations-oid anymore since its tenant specific and cant be automated.

Reply:
@dkirby-ms yep. @alihhussain refer to #765